azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -180,6 +197,9 @@ def run(self, goal_state, ext_gs_updated):
             self._updater.retrieve_agent_version(agent_family, goal_state)
 
             if not self._updater.is_retrieved_version_allowed_to_update(agent_family):
+                # Reset the last error report msg  when this condition met as there is no issue with update since requested
+                # version is same as current version or below than daemon version. In that case, we shouldn't report any error msg in the status report
+                self._last_attempted_update_error_msg = ""
```

Comment:
Should we also reset the _last_attempted_update_error_msg in the condition that update is not allowed right now (line 194)?